model/common/src/icon4py/model/common/grid/vertical.py

```diff
@@ -726,7 +726,9 @@ def _check_flatness_of_flat_level(
     array_ns: ModuleType = np,
 ) -> None:
     # Check if level nflatlev is still flat
-    if not array_ns.all(vertical_coordinate[:, nflatlev - 1] == vct_a[nflatlev - 1]):
+    if not array_ns.all(
+        vertical_coordinate[:, np.maximum(0, nflatlev - 1)] == vct_a[np.maximum(0, nflatlev - 1)]
```

Comment:
When comparing simple `int`s, there is not need to use numpy, I would just the python function:
```suggestion
        vertical_coordinate[:, max(0, nflatlev - 1)] == vct_a[max(0, nflatlev - 1)]
```